crates/hyperswitch_domain_models/src/payment_method_data.rs

```diff
@@ -135,10 +135,37 @@ pub struct MifinityData {
 }
 
 #[derive(Eq, PartialEq, Clone, Debug, serde::Deserialize, serde::Serialize)]
-
+#[serde(rename_all = "snake_case")]
 pub struct SamsungPayWalletData {
-    /// The encrypted payment token from Samsung
-    pub token: Secret<String>,
+    pub payment_credential: SamsungPayWalletCredentials,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "snake_case")]
+pub struct SamsungPayWalletCredentials {
+    pub method: Option<String>,
+    /// Value if credential is enabled for recurring payment
```

Comment:
You can ignore doc comments for domain models, if we're not gonna bother maintaining them and keeping them up-to-date.